iotdb-client/session/src/main/java/org/apache/iotdb/session/Session.java

```diff
@@ -905,6 +906,25 @@ private SessionDataSet executeStatementMayRedirect(String sql, long timeoutInMs)
     }
   }
 
+  private SessionConnection getQuerySessionConnection() {
+    Optional<TEndPoint> endPoint =
+        availableNodes == null ? Optional.empty() : availableNodes.getQueryEndPoint();
+    if (!endPoint.isPresent()) {
+      return defaultSessionConnection;
+    }
+    SessionConnection connection =
+        endPointToSessionConnection.computeIfAbsent(
```

Comment:
If both `enableRedirection` and `enableQueryRedirection` are set to false, this will cause an NPE since `endPointToSessionConnection` has not been initialized.

Reply:
fixed